eureka/S2_calibrations/s2_calibrate.py

```diff
@@ -141,6 +161,160 @@ def calibrateJWST(eventlabel, ecf_path='./', s1_meta=None):
+        the input files to use.
+
+    Notes
+    -------
+    History:
+
+    - April 15, 2022 Taylor Bell
+        Initial version.
+    '''
+    rootdir = os.path.join(meta.topdir, *meta.inputdir.split(os.sep))
+    if rootdir[-1]!='/':
+        rootdir += '/'
+    fnames = glob.glob(rootdir+'*'+meta.suffix + '.fits')
+    if len(fnames)==0:
+        # There were no rateints files in that folder, so let's see if there are in children folders
+        fnames = glob.glob(rootdir+'**/*'+meta.suffix + '.fits', recursive=True)
+        fnames = sn.sort_nicely(fnames)
+
+    if len(fnames)==0:
+        # If the code can't find any of the reqested files, raise an error and give a helpful message
+        raise AssertionError(f'Unable to find any "{meta.suffix}.fits" files in the inputdir: \n"{meta.inputdir}"!\n'+
+                             f'You likely need to change the inputdir in {meta.filename} to point to the folder containing the "{meta.suffix}.fits" files.')
+    
+    folders = np.unique([os.sep.join(fname.split(os.sep)[:-1]) for fname in fnames])
+    if len(folders)>=1:
+        # get the file with the latest modified time
+        folder = max(folders, key=os.path.getmtime)
+    
+    if len(folders)>1:
+        # There may be multiple runs - use the most recent but warn the user
+        print(f'WARNING: There are multiple folders containing "{meta.suffix}.fits" files in your inputdir: \n"{meta.inputdir}"\n'
+             +f'Using the files in: \n{folder}\n'
+              +'and will consider aperture ranges listed there. If this metadata file is not a part\n'
+              +'of the run you intended, please provide a more precise folder for the metadata file.')
+
+    meta.inputdir = folder[len(meta.topdir):]
+    if meta.inputdir[-1] != '/':
+        meta.inputdir += '/'
+
+    return meta
+
+def load_general_s1_meta_info(meta, ecf_path, s1_meta):
+    '''Loads in the S1 meta save file and adds in attributes from the S2 ECF.
+
+    Parameters
+    ----------
+    meta:    MetaClass
+        The new meta object for the current S2 processing.
+    ecf_path:
+        The absolute path to where the S2 ECF is stored.
+
+    Returns
+    -------
+    meta:   MetaClass
+        The S1 metadata object with attributes added by S2.
+
+    Notes
+    -------
+    History:
+
+    - April 15, 2022 Taylor Bell
+        Initial version.
+    '''
+    # Need to remove the topdir from the outputdir
+    s1_outputdir = s1_meta.outputdir[len(meta.topdir):]
+    if s1_outputdir[0]=='/':
+        s1_outputdir = s1_outputdir[1:]
+    if s1_outputdir[-1]!='/':
+        s1_outputdir += '/'
+    s1_topdir = s1_meta.topdir
+    
+    # Load S2 Eureka! control file and store values in the S1 metadata object
+    ecffile = 'S2_' + meta.eventlabel + '.ecf'
+    meta = s1_meta
+    meta.read(ecf_path, ecffile)
+
+    # Overwrite the inputdir with the exact output directory from S1
+    meta.inputdir = os.path.join(s1_topdir, s1_outputdir)
```

Comment:
I don't like this implementation, both here and at other stages. The inputdir should use s2_inputdir from the S2 ECF file. Otherwise, there's no way to read in files that were moved or from a renamed directory.

Reply:
We can discuss this on Monday, but the issue is that I think we should allow autocompleted paths based on the inputdir. For example, it is far more convenient to specify `inputdir = Stage1` and have Stage 2 find the newest directory in that folder. Before this PR, one would have to run Stage 1, see what foldername the FITS files ended up being put into, copy-paste that foldername into their Stage 2 ECF, and then run Stage 2 - an annoying step for Stage 1->2, but something that would get super tedious at other stages where many folders might be made.

A better example is Stage 3->4, where we wanted to keep the ability for Stage 4 to consider all of the apertures used in Stage 3. So Stage 3 outputs Stage3/S3_2022-03-21_miri_lrs_ap2_bg4_run1 and Stage3/S3_2022-03-21_miri_lrs_ap2_bg14_run1 as an example, and then Stage 4 somehow needs to find those two particular folders (with potentially many other folders inside the Stage3 folder from previous attempts). The code that we have now will look for Meta_Save.dat files in Stage 4, find none, look inside all the folders inside the Stage4 folder and find two Meta_Save.dat files (or many more if there are older runs) and by default will use the most recently created/edited one. But if you set `allapers=True` in Stage4, the code needs to find the other apertures that were used in that same Stage 3 run, so it uses the one Stage 3 Meta_Save file that it does have to figure out where the other apertures would have been saved.

It's all kind of a mess I know, but given the I/O setup we have presently it was the best I could come up with and has been super convenient as long as you're not moving the files to a different drive or sharing them with someone else.  While somewhat uncommon, I agree it's important to support the ability to move the files around at will, and I think we should take the opportunity to rework all the connections between Stages at the same time as we're changing the pipeline's I/O. One option might be putting all the apertures for a Stage3 run into one folder (e.g. Stage3/S3_2022-03-21_miri_lrs_run1/ap2bg4 and Stage3/S3_2022-03-21_miri_lrs_run1/ap2bg14).

Either way, edits to these functions is going to significantly affect all the stages, and I think doing that at around the same time as changing the I/O to Astreus would make for a clean transition. This was just meant as a quick and easy fix to the Stage 1->2 connection to make it consistent with all the other stages for the time being.

Reply:
Ok, let's discuss on Monday.  I agree that we need an overhaul, in part because I frequently rename my folders to be more descriptive than simply "runX".  Your proposed solution (Stage3/S3_2022-03-21_miri_lrs_run1/ap2bg4) should work for both our needs, so long as meta.inputdir is derived from the S4 ecf file (e.g., inputdir=Stage3/S3_2022-03-21_miri_lrs_run1).